src/components/ProductCard.tsx

```diff
@@ -0,0 +1,103 @@
+import StarIcon from '@mui/icons-material/Star'
+import { FiMinus as Minus } from 'react-icons/fi'
+import { Rating, styled } from '@mui/material'
+import Image from 'next/image'
+import { GiBottleCap as Caps } from 'react-icons/gi'
+import { LiaPercentSolid as Percent } from 'react-icons/lia'
+import { useState } from 'react'
+
+const StyledRating = styled(Rating)({
+	'& .MuiRating-iconFilled': {
+		color: '#f1f5f9',
+		filter: 'drop-shadow(1px 0.75px 0px rgb(0 0 0 / 1))',
+	},
+	'& .MuiRating-iconEmpty': {
+		color: '#1e293b',
+		fill: 'black',
+	},
+})
+
+const ProductCard = () => {
+	const [isOnSale, setisOnSale] = useState(true)
```

Comment:
Is this dispatcher called?

Reply:
No but Its outdated and now `zustand` handling this